### PR TITLE
LIVE-1250: Split out AR and ER renderers

### PR DESCRIPTION
**Pull request by faresite**
## Why are you doing this?

We use a `render` function to render blocks, now that some designs will diverge for ER (pullquote for example). We need two separate renderers, however a lot of the element renderers have been refactored out so we don't have to duplicate their code if the element is the same.

## Changes

- Refactor out element renderers that are > 1 line
- Created `renderEditions` and `renderEditionsAll`
- typed three elements: `Text`, `Embed` and `Instagram`

### CODE REVIEW

File: src/renderer.ts [hunk 1]
```diff
@@ -492,226 +752,148 @@ const render = (format: Format, excludeStyles = false) => (
...
-				element.alt,
-				map((alt) => figure(alt)),
-				withDefault(figure(null)),
-			);
+		case ElementKind.Instagram:
+			return instagramRenderer(element);
+
+		case ElementKind.ExplainerAtom: {
+			return h(ExplainerAtom, { ...element });
 		}
```
Comment (JamieB-gu) [inline src/renderer.ts:763]:
```suggestion
		case ElementKind.ExplainerAtom:
			return h(ExplainerAtom, { ...element });
```

File: src/renderer.ts [hunk 2]
```diff
@@ -492,226 +752,148 @@ const render = (format: Format, excludeStyles = false) => (
...
-			return h('div', props);
+		case ElementKind.ProfileAtom:
+			return profileAtomRenderer(format, element);
+
+		case ElementKind.TimelineAtom:
+			return timelineAtomRenderer(format, element);
+
+		case ElementKind.ChartAtom: {
+			return h(ChartAtom, { ...element });
 		}
```
Comment (JamieB-gu) [inline src/renderer.ts:779]:
```suggestion
		case ElementKind.ChartAtom:
			return h(ChartAtom, { ...element });
```

File: src/renderer.ts [hunk 3]
```diff
@@ -492,226 +752,148 @@ const render = (format: Format, excludeStyles = false) => (
...
+	switch (element.kind) {
+		case ElementKind.Text:
+			return textRenderer(format, excludeStyles, element);
+
+		case ElementKind.Image:
+			return imageRenderer(format, element, key);
+
+		case ElementKind.Pullquote: {
+			const { quote, attribution } = element;
+			return h(Pullquote, { quote, attribution, format, key });
```
Comment (JamieB-gu) [inline src/renderer.ts:808]: Is the next step to build an `EditionsPullquote` component?

File: src/renderer.ts [hunk 4]
```diff
@@ -492,226 +752,148 @@ const render = (format: Format, excludeStyles = false) => (
...
-				},
-				dislikeHandler: () => {
-					console.log('dislike clicked');
-				},
-				expandCallback: () => {
-					console.log('expand clicked');
-				},
-			});
+		case ElementKind.LiveEvent: {
+			return h(LiveEventLink, { ...element, key });
```
Comment (JamieB-gu) [inline src/renderer.ts:817]: I think you can drop the enclosing `{}` here (it won't let me add the suggestion).

**Review by JamieB-gu — APPROVED** [review]
Excellent! This is the direction I've been hoping we'd go in ⬆️

File: src/renderer.ts (hunk 3 above)
Comment (faresite) [reply src/renderer.ts:808]: Indeed 👍

**Pull request merged**